backend/src/bundles/wallets/wallets.controller.ts

```diff
@@ -0,0 +1,338 @@
+     *         content:
+     *           application/json:
+     *             schema:
+     *               $ref: '#/components/schemas/Wallet'
+     */
+
+    private async create(
+        request: TokenRequestDto,
+    ): Promise<ApiHandlerResponse> {
+        if (!request.token) {
+            throw new Error(WalletValidationMessage.TOKEN_REQUIRE);
+        }
+        const userId = getUserIdFromToken(request.token);
+        const wallet = await this.walletService.newCreate(request.body, userId);
+        return {
+            status: HttpCode.OK,
+            payload: wallet,
+        };
+    }
+
+    /**
+     * @swagger
+     * /wallets/{id}:
+     *   put:
+     *     tags: [Wallets]
+     *     summary: Update a wallet
+     *     description: Update a wallet's name, currency ID, and balance.
+     *     security:
+     *       - bearerAuth: []
+     *     parameters:
+     *       - name: id
+     *         in: path
+     *         description: ID of the wallet to update.
+     *         required: true
+     *         schema:
+     *           type: string
+     *         example: 4b7908cc-5581-4d74-8910-7f2bba8cb49b
+     *     requestBody:
+     *       required: true
+     *       content:
+     *         application/json:
+     *           schema:
+     *             type: object
+     *             properties:
+     *               name:
+     *                 type: string
+     *                 description: Name of the wallet.
+     *               currencyId:
+     *                 type: string
+     *                 description: ID of the currency used in the wallet.
+     *                 format: uuid
+     *               balance:
+     *                 type: number
+     *                 description: Balance of the wallet.
+     *                 format: float
+     *     responses:
+     *       '200':
+     *         description: Wallet updated successfully.
+     *         content:
+     *           application/json:
+     *             schema:
+     *               $ref: '#/components/schemas/Wallet'
+     *     securitySchemes:
+     *       bearerAuth:
+     *         type: http
+     *         scheme: bearer
+     *         bearerFormat: JWT
+     * components:
+     *   schemas:
+     *     Wallet:
+     *       type: object
+     *       properties:
+     *         id:
+     *           type: string
+     *           format: uuid
+     *         name:
+     *           type: string
+     *         currencyId:
+     *           type: string
+     *           format: uuid
+     *         balance:
+     *           type: number
+     *           format: float
+     */
+
+    private async update(
+        request: TokenRequestDto,
+    ): Promise<ApiHandlerResponse> {
+        if (!request.token) {
+            throw new Error(WalletValidationMessage.TOKEN_REQUIRE);
+        }
+        const userId = getUserIdFromToken(request.token);
+        const newWallet = await this.walletService.newUpdate(
```

Comment:
```suggestion
        const updatedWallet = await this.walletService.update(
```

Reply:
Why are you using 'new' prefix?

Reply:
on methods im using prefix 'new' because original methods in parents interface have another arguments thats why i have a lot of conflicts if i use the original name of method

Reply:
I updated the names in a latest commit